app/design/frontend/base/default/template/oyst/oneclick.phtml

```diff
@@ -23,49 +23,20 @@ $oneClickUrl = $this->getOneClickUrl();
 /** @var Oyst_OneClick_Helper_Catalog_Data $catalogHelper */
 $catalogHelper = Mage::helper('oyst_oneclick/catalog_data');
 
+$isProductAddtocartFormValidate = Mage::getStoreConfig("oyst/oneclick/product_addtocart_form_validate");
+
 if ($catalogHelper->isSupportedProduct($product)) :
 ?>
     <div id="oyst-1click-button"></div>
     <script>
         (function () {
-            window.__OYST__ = window.__OYST__ || {};
-            window.__OYST__.getOneClickURL = function (cb, opts) {
-                opts = opts || {};
-
-                ready(function () {
-                    var qty = $$("input[name='qty']")[0].value ? $$("input[name='qty']")[0].value : 1;
-
-                    var form = new FormData();
-                    form.append("productRef", <?php echo $product->getId(); ?>);
-                    form.append("quantity", qty);
-                    form.append("variationRef", getSimpleProductId());
-
-                    var settings = {
-                        async: true,
-                        cacheControl: "no-cache",
-                        data: form,
-                        method: "POST",
-                        mimeType: "multipart/form-data",
-                        url: "<?php echo $oneClickUrl ?>"
-                    };
-
-                    var xhr = new XMLHttpRequest();
-                    xhr.open(settings.method, settings.url, settings.async);
-                    xhr.setRequestHeader('cache-control', settings.cacheControl);
-                    xhr.onload = function () {
-                        if (200 === xhr.status) {
-                            var data = JSON.parse(xhr.responseText);
-                            var isErrorsInForm = null;
-                            // If not the preload of button run form validation to know if they are errors
-                            if (!opts.preload) {
-                                var isErrorsInForm = !isOystOneClickButtonFormValid('product_addtocart_form');
-                            }
-                            cb(isErrorsInForm, data.url);
-                        }
-                    };
-                    xhr.send(form);
-                });
-            };
+            if ("function" === typeof oystOneClick ) {
+                oystOneClick(
+                    <?php echo $product->getId() ?>,
```

Comment:
ce n'est pas un string ?

Reply:
![image](https://user-images.githubusercontent.com/861701/31433819-caebdb60-ae7a-11e7-9031-16507f355e0f.png)
Je test la présente de la fonction

Reply:
Why not just `if(oystOneClick)` ?

Reply:
we are probably not talking about the same line.
i wondered why there are no quotes around `<?php echo $product->getId() ?>`.
Isn't it supposed to be a string?

Reply:
@jney of course product_reference is a string so I can surround it with quote.

Reply:
@nicolaschenet if I do it with `if(oystOneClick)`and it's not available I'll get an error `Uncaught ReferenceError: oystOneClick is not defined    at <anonymous>:1:1` is that a goo practice ?